### PR TITLE
merged parabolic1d.cpp, parabolic1d.m, and makefile to reflect the new directory structure that is in the master branch. 

### LINKED ISSUE #14: Implement decomposition in parabolic1.m
Implement the change given in the following lines in examples_MATLAB/parabolic1d.m

https://github.com/csrc-sdsu/mole/blob/c78ef4680da883d8a7f98b0d93ab77f4248ab572/examples/examples_MATLAB/parabolic1D.m#L63-L65

### PR DESCRIPTION
**Pull request by aneeshs1729**
this submission adds a parabolic1d.cpp example, adds decomposition to  parabolic1d.m , and changes the makefile for examples cpp accordingly. Will resolve #14  #24 .

### CODE REVIEW

**Review by aneeshs1729 — COMMENTED** [review]
@valeriabarra - I have created this pull request after hand comparing the files between the master branch and my branch. 

The changes in the master branch are reflected in my files now

File: examples/cpp/Makefile [hunk 1]
```diff
@@ -1,17 +1,23 @@
 MOLE = ../../src/cpp
 
-CXX ?= g++
+#ARMA = /Users/aneeshsrinivas/Documents/GitHub/armadillo-14.0.3
```
Comment (jbrzensk) [inline examples/cpp/Makefile:3]: Remove all of these extra "macOS" comments and commented code. It is confusing, as some macOS users might have libraries there.

File: examples/cpp/Makefile [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+CXX ?= g++ #for non mac users.
 
 ifeq ($(DEBUG),1)
 CXXFLAGS = -O0 -g
 else
 CXXFLAGS = -O3
 endif
-CXXFLAGS += -fopenmp -DARMA_DONT_USE_WRAPPER
+CXXFLAGS += -fopenmp -DARMA_DONT_USE_WRAPPER #for non mac users.
+#CXXFLAGS +=  -DARMA_DONT_USE_WRAPPER #for mac os
```
Comment (jbrzensk) [inline examples/cpp/Makefile:14]: Remove the comments on these lines as well. Some macOS users might have openmp installed.

File: examples/cpp/Makefile [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
 
-INCPATH = -I. $(if $(ARMA), -I$(ARMA)/include) -I$(MOLE)
+INCPATH = -I. $(if $(ARMA), -I$(ARMA)/include) -I$(MOLE) #for non mac users.
```
Comment (jbrzensk) [inline examples/cpp/Makefile:16]: Remove the macOS comments

File: examples/cpp/Makefile [hunk 4]
```diff
@@ -26,13 +32,17 @@ CXXFLAGS += -DEIGEN
...
 else
-LIBS += -lsuperlu
+LIBS += -lsuperlu #for non mac users.
+#LIBS += -lsuperlu -L/opt/homebrew/Cellar/superlu/7.0.0/lib #for mac os
 endif
 endif
 
-all: transport1D schrodinger1D elliptic1D elliptic2D convection_diffusion RK2
+
+all: transport1D schrodinger1D elliptic1D elliptic2D parabolic1D convection_diffusion RK2
```
Comment (jbrzensk) [inline examples/cpp/Makefile:44]: Everything from here down looks good. :)

File: examples/cpp/parabolic1D.cpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    *Note that in this example dt=dx^2/3 like it is in the matlab example. 
+    */
+    Laplacian L(k,m,dx); 
+    vec solution(m + 2); 
+    solution(0)=100;
+    solution(m+1)=100;
+    vec k1(m+2); 
+    double t=t0;
+    while (t <= tf) { //time integration with euler method. 
+      k1=L*(solution);
```
Comment (jbrzensk) [inline examples/cpp/parabolic1D.cpp:29]: nit: indentation, two spaces instead of four

File: examples/cpp/parabolic1D.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      solution=solution+dt*k1;
+      t=t+dt;
+    }
+
+    std::cout << solution;
+
+
```
Comment (jbrzensk) [inline examples/cpp/parabolic1D.cpp:36]: nit: extra line

File: examples/matlab/parabolic1D.m [hunk 7]
```diff
@@ -49,9 +49,13 @@
 else
     tic
     % Implicit
-    L = -alpha*dt*L + speye(size(L));
-    
-    % Time integration loop
+    L = -alpha*dt*L + speye(size(L)); 
+    dL=decomposition(L)
+    %This line precomputes the LU decomposition of L and stores it as a
```
Comment (jbrzensk) [inline examples/matlab/parabolic1D.m:54]: YES!!!!

**Review by jbrzensk — CHANGES_REQUESTED** [review]
Noted the things we talked about in my office yesterday.

File: examples/cpp/Makefile [hunk 8]
```diff
@@ -1,17 +1,18 @@
 MOLE = ../../src/cpp
 
-CXX ?= g++
+#ARMA = /Users/aneeshsrinivas/Documents/GitHub/armadillo-14.0.3
```
Comment (jbrzensk) [inline examples/cpp/Makefile:3]: Line 3 is a reference to your ARMA. Please put it back to the old one.

**Review by jbrzensk — APPROVED** [review]

File: examples/cpp/Makefile (hunk 4 above)
Comment (aneeshs1729) [reply examples/cpp/Makefile:37]: Hello @jbrzensk is there any other changes i need to make?

**Pull request merged**